lib/github_service/commands/run_test.rb

```diff
@@ -0,0 +1,278 @@
+module GithubService
+  module Commands
+    # = GithubService::Commands::RunTest
+    #
+    # Triggers a build given the configured test repo:
+    #
+    #   Settings.github.run_tests_repo.*
+    #
+    # Which doing so will:
+    #
+    #    - validate the command
+    #    - create a branch
+    #    - create a commit with the travis.yml changes
+    #    - push said branch to the origin
+    #    - create a pull request for that branch
+    #
+    # More info can be found here on how the whole process works:
+    #
+    #   https://github.com/ManageIQ/manageiq-cross_repo-tests
+    #
+    # == Command structure
+    #
+    # @miq-bot run-tests [<repos-to-test>] [including <extra-repos>]
+    #
+    # where:
+    #
+    #   - a `repo` is of the form [org/]repo[@ref|#pr]
+    #   - `repos-to-test` is a list of repos to have tested
+    #   - `extra-repos` is a list of repos (gems) to override in the bundle
+    #
+    # each "lists of repos" should be comma delimited.
+    #
+    # == Example
+    #
+    # In ManageIQ/manageiq PR #1234
+    #
+    #   @miq-bot run-tests manageiq-api,manageiq-ui-classic#5678 \
+    #     including Fryguy/more_core_extensions@feature,Fryguy/linux_admin@feature
+    #
+    # will create a commit with the .travis.yml changes:
+    #
+    #   @@ -14,6 +14,6 @@ matrix:
+    #      fast_finish: true
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=Fryguy/more_core_extensions@feature,Fryguy/linux_admin@feature,ManageIQ/manageiq#1234
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=ManageIQ/manageiq-api
+    #   +  - TEST_REPO=ManageIQ/manageiq-ui-classic#5678
+    #
+    # TODO:  Handle the "self" case, where `manageiq` is also a TEST_REPO
+    #
+    # (maybe include a "self" helper as well?)
+    #
+    class RunTest < Base
+      # Reference to the branch we are creating off of origin/master
+      attr_reader :branch_ref
+
+      # The user calling the command
+      attr_reader :issuer
+
+      # The (extra) repo(s) being targeted to be included in the test run
+      attr_reader :repos
+
+      # The repo(s) that will have the test suite run
+      attr_reader :test_repos
+
+      # The *-cross_repo-tests rugged instance
+      attr_reader :rugged_repo
+
+      # The arguments for the `run-test` command being called
+      attr_reader :value
+
+      restrict_to :organization
+
+      def self.test_repo_url
+        Settings.github.run_tests_repo.url
+      end
+
+      def self.test_repo_name
+        Settings.github.run_tests_repo.name
+      end
+
+      def self.test_repo_clone_dir
+        @test_repo_clone_dir ||= begin
+                                   url_parts = test_repo_url.split("/")[-2, 2]
+                                   repo_org  = url_parts.first
+                                   repo_dir  = test_repo_name
+                                   File.join(::Repo::BASE_PATH, repo_org, repo_dir)
```

Comment:
There is now a class method `Repo.path` that gives you the path if you give it a qualified repo name (e.g. "SomeOrg/some_repo"), so that might be perfect here.

Reply:
It will save about 5 characters...

```ruby
::Repo.path(File.join(repo_org, repo_dir))
```

Thought it was only going to be 1 characters, but it turns out that `Rugged::Repository` will take a `Pathname` instance, so `.to_s` wouldn't be required.

Though, what I will change it to is:

```ruby
::Repo::BASE_PATH.join repo_org, repo_dir
```

Since `BASE_PATH` is a `Pathname` and not a string.

Reply:
Correction:  The above still needs to be a `String` since `Rugged::Repository.clone_at` _**does**_ require a string.